File: js/services/ReportService.js
app.service('ReportService', function($http){
    
    this.lastEvaluateOfServiceProvider = function(serviceProvider, service, limit, callback, error) {
        var token = window.localStorage['token'];
		$http.get('http://localhost:8080/WebService/report/lastEvaluateOfServiceProvider?token=' + token +
         '&serviceProvider=' + serviceProvider + '&service=' + service + '&limit=' + limit).
		then(callback, error);
    };
    
    this.lastEvaluateOfServiceInNetwork = function(serviceProvider, service, limit, callback, error) {
        var token = window.localStorage['token'];
		$http.get('http://localhost:8080/WebService/report/lastEvaluateOfServiceInNetwork?token=' + token +
         '&serviceProvider=' + serviceProvider + '&service=' + service + '&limit=' + limit).
		then(callback, error);
    };
    
    this.lastEvaluate = function(service, limit, callback, error) {
        var token = window.localStorage['token'];
		$http.get('http://localhost:8080/WebService/report/lastEvaluate?token=' + token +
                  '&service=' + service + '&limit=' + limit).
		then(callback, error);
    };
    
    this.lastEvaluateInMyCity = function(service, limit, callback, error) {
        var token = window.localStorage['token'];
		$http.get('http://localhost:8080/WebService/report/lastEvaluateInMyCity?token=' + token +
                '&service=' + service + '&limit=' + limit).
		then(callback, error);
    };
});

File: entrega-cd/app/back-end/webservice-java/src/br/edu/univas/si/tcc/trunp/api/ReportService.java
package br.edu.univas.si.tcc.trunp.api;

import org.codehaus.jettison.json.JSONException;
import org.codehaus.jettison.json.JSONObject;

public interface ReportService {
	
	public JSONObject lastEvaluateOfServiceProvider(String token, String serviceProvider, String service, int limit) throws JSONException;
	
	public JSONObject lastEvaluateOfServiceInNetwork(String token, String serviceProvider, String service, int limit) throws JSONException;

	public JSONObject lastEvaluate(String serviceProvider, String service, int limit) throws JSONException;
	
	public JSONObject lastEvaluateInMyCity(String serviceProvider, String service, int limit) throws JSONException;
	
}
